service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/mapper/PatientMapper.java

```diff
@@ -1,19 +1,167 @@
+import org.hl7.fhir.dstu3.model.ContactPoint;
+import org.hl7.fhir.dstu3.model.Enumerations;
+import org.hl7.fhir.dstu3.model.Extension;
+import org.hl7.fhir.dstu3.model.HumanName;
 import org.hl7.fhir.dstu3.model.Patient;
+import org.hl7.fhir.dstu3.model.Reference;
+import org.hl7.fhir.dstu3.model.StringType;
+import org.hl7.fhir.dstu3.model.Organization;
+
 import org.springframework.stereotype.Component;
 
+import uk.nhs.connect.iucds.cda.ucr.POCDMT000002UK01LanguageCommunication;
+import uk.nhs.connect.iucds.cda.ucr.POCDMT000002UK01Patient;
+import uk.nhs.connect.iucds.cda.ucr.POCDMT000002UK01PatientRole;
+import uk.nhs.adaptors.oneoneone.cda.report.util.NodeUtil;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.hl7.fhir.dstu3.model.IdType.newRandomUuid;
+
 @Component
+@AllArgsConstructor
 public class PatientMapper {
-    //TODO
-    // This is just a placeholder for the mapper. EpisodeOfCare requires a reference to a patient for message to be valid
 
-    public Patient mapPatient() {
-        Patient patient = new Patient();
-        patient.setId(newRandomUuid());
-        return patient;
+    private final AddressMapper addressMapper;
+    private final ContactPointMapper contactPointMapper;
+    private final PeriodMapper periodMapper;
+    private final GuardianMapper guardianMapper;
+    private final HumanNameMapper humanNameMapper;
+    private final OrganizationMapper orgMapper;
+
+    public Patient mapPatient(POCDMT000002UK01PatientRole patientRole) {
+        Patient fhirPatient = new Patient();
+        fhirPatient.setIdElement(newRandomUuid());
+        if (patientRole.isSetPatient()) {
+            POCDMT000002UK01Patient itkPatient = patientRole.getPatient();
+            fhirPatient.setActive(true);
+            fhirPatient.setName(getNames(itkPatient));
+            fhirPatient.setAddress(getAddresses(patientRole));
+            fhirPatient.setTelecom(getContactPoints(patientRole));
+
+            fhirPatient.addGeneralPractitioner(getGeneralPractioner(patientRole));
+
+            if (itkPatient.sizeOfLanguageCommunicationArray() > 0) {
+                Stream.of(itkPatient.getLanguageCommunicationArray())
+                    .map(this::getLanguageCommunicationCode)
+                    .forEach(fhirPatient::setLanguage);
+            }
+
+            fhirPatient.setContact(getContactComponents(itkPatient));
+            fhirPatient.setExtension(getExtensions(itkPatient));
+
+            if (itkPatient.isSetBirthTime()) {
+                fhirPatient.setBirthDate(periodMapper.mapPeriod(itkPatient.getBirthTime()).getStart());
+            }
+
+            if (itkPatient.isSetAdministrativeGenderCode()) {
+                String displayName = itkPatient.getAdministrativeGenderCode().getDisplayName();
+                if (displayName == null) displayName = "unknown";
```

Comment:
fhirPatient.setGender(Enumerations.AdministrativeGender
                    .fromCode(displayName == null ? "unknown" : displayName.toLowerCase()));
just a suggestion :)

Reply:
Effected in the module - it was a good idea and easier to read.